### PR TITLE
Merge Beta 1.1.3 -> main

### PR DESCRIPTION
**Pull request by michal-zurkowski**
Dev pool is now selected from the list.
Show new statistic info about hashrate after each new block.
Another attempt to fix stratum issues from some users.
Include prefetch for loading / storing data for Cryptonight (~1-2% improvement)
Rewrite 2way Cryptonight. (~15-50% improvement)
Include 2 new options for AVX2+:
-- --tune - Optimal tuning for 2way or 1way solving of Cryptonight. Takes ~34 minutes. Creates tune_config file that can be used later. It is recommended to run it with --benchmark to disable any stratum and network communication. Can be used directly with your normal mining options and will start mining after the tuning process. After doing it once, use --tune-config after that.
-- --tune-config=FILE - Use tuning config generated by --tune` while mining.

### CODE REVIEW

**Review by WyvernTKC — CHANGES_REQUESTED** [review]
Your BUTK address is missing. Can you update this?

File: cpu-miner.c [hunk 1]
```diff
@@ -2677,16 +2738,18 @@ char *rpc_url_original = NULL;
...
 // idx 0 - Ausminer
 // idx 1 - Delgon
-const uint8_t max_idx = 2;
+const uint8_t max_idx = 3;
 uint8_t donation_url_idx[2] = {0, 0};
 char *donation_url[2][3] = {
-    {"stratum+tcp://r-pool.net:3008", "stratum+tcp://rtm.suprnova.cc:6273"},
-    {"stratum+tcp://r-pool.net:3008", "stratum+tcp://rtm.suprnova.cc:6273"}};
+    {"stratum+tcp://rtm.suprnova.cc:6273", "stratum+tcp://r-pool.net:3008",
+     "stratum+tcp://rtm.ausminers.com:3333"},
```
Comment (WyvernTKC) [inline cpu-miner.c:2745]: Can you update the ports for Ausminers to 3001 or 3002 or 3003

File: cpu-miner.c [hunk 2]
```diff
@@ -2677,16 +2738,18 @@ char *rpc_url_original = NULL;
...
+    {"stratum+tcp://rtm.suprnova.cc:6273", "stratum+tcp://r-pool.net:3008",
+     "stratum+tcp://rtm.ausminers.com:3333"}};
 char *donation_userRTM[2] = {"RXq9v8WbMLZaGH79GmK2oEdc33CTYkvyoZ",
                              "RQKcAZBtsSacMUiGNnbk3h3KJAN94tstvt"};
 char *donation_userBUTK[2] = {"XdFVd4X4Ru688UVtKetxxJPD54hPfemhxg",
                               "XdFVd4X4Ru688UVtKetxxJPD54hPfemhxg"};
```
Comment (WyvernTKC) [inline cpu-miner.c:2751]: Add your BUTK address

File: cpu-miner.c [hunk 3]
```diff
@@ -2092,6 +2089,65 @@ static void stratum_gen_work(struct stratum_ctx *sctx, struct work *g_work) {
   // Update data and calculate new estimates.
   if ((stratum_diff != sctx->job.diff) ||
       (last_block_height != sctx->block_height)) {
+    double hashrate = 0.;
```
Comment (WyvernTKC) [inline cpu-miner.c:2092]: Really like this change

**Review by WyvernTKC — APPROVED** [review]

File: algo/gr/cryptonote/cryptonight.c [hunk 4]
```diff
@@ -379,281 +456,131 @@ void cryptonight_turtlelite_hash(const void *input, void *output) {
...
-    d = _mm_aesenc_si128(d, *R128(&k[5]));
-    d = _mm_aesenc_si128(d, *R128(&k[6]));
-    d = _mm_aesenc_si128(d, *R128(&k[7]));
-    d = _mm_aesenc_si128(d, *R128(&k[8]));
-    d = _mm_aesenc_si128(d, *R128(&k[9]));
-    _mm_storeu_si128((R128(out + i * AES_BLOCK_SIZE)), d);
-  }
-}
+// Requires 2x memory allocated in hp_state.
+__attribute__((always_inline)) void
```
[the remark this answers is not in the document]
Comment (michal-zurkowski) [reply algo/gr/cryptonote/cryptonight.c:460]: Doing 2 hashes at the same time with interleaving parts in the main iteration loop makes it possible to prefetch some of the data in advance which is not really possible in the cryptonight_hash(). This leads to an increase in performance if the L3 cache on the CPU is adequate and creates an alternative method that might be more beneficial for some CPUs compared to calling cryptonight_hash() twice.

**Pull request merged**